onnx_chainer/functions/array.py

```diff
@@ -52,10 +52,85 @@ def convert_Depth2Space(func, onnx_op_name, opset_version, input_names,
+        # axis means the index of input x, adjust None and Ellipsis counts
+        axis = i - len(unsqueeze_idxs) + skipped
+        if isinstance(idx, slice):
+            if idx.step is not None and idx.step != 1:
+                raise ValueError(
+                    'GetItem with {}step slicing is not supported in ONNX '
+                    'Slice operator'.format(idx.step))
+            axes.append(axis)
+            starts.append(idx.start)
+            ends.append(idx.stop)
+        elif isinstance(idx, int):
+            axes.append(axis)
+            starts.append(idx)
+            ends.append(idx+1)
+            squeeze_idxs.append(axis)
+        elif idx is None:
+            unsqueeze_idxs.append(i - len(squeeze_idxs) + skipped)
+        elif idx is Ellipsis:
+            # calculate rest slice number except None, GetItem does not allow
+            # multiple Ellipsis, so ignore latter Ellipsis count
+            rest_slice_len = len(
+                [idx_ for idx_ in func.slices[i+1:] if idx_ is not None])
+            for ellipsis_i in range(axis, len(x.shape)-rest_slice_len):
+                axes.append(ellipsis_i)
+                starts.append(0)
+                ends.append(x.shape[ellipsis_i])
+                skipped += 1
+            skipped -= 1  # for next loop
+        else:
+            # not support advanced index like `array[[0,1], [0, 1]]`
+            raise ValueError(
+                'GetItem with type {} cannot handle in ONNX Slice, so that '
+                'ONNX-Chainer does not accept the type'.format(type(idx)))
+    # GetItem collects rest axes on default, needs to setup for Slicing
+    sliced_axes_len = len(axes)
+    if sliced_axes_len < len(x.shape):
+        for i, length in enumerate(x.shape[sliced_axes_len:]):
+            axes.append(i+sliced_axes_len)
+            starts.append(0)
+            ends.append(length)
+    nodes = []
+    slice_out_names = output_names
+
+    if unsqueeze_idxs:
+        unsqueeze_object = object()
+        dummy_objects.append(unsqueeze_object)
+        unsqueeze_object_name = str(id(unsqueeze_object))
+        slice_out_names = [unsqueeze_object_name]
+        nodes.append(helper.make_node(
+            'Unsqueeze', [unsqueeze_object_name], output_names,
+            axes=unsqueeze_idxs))
+
+    if squeeze_idxs:
+        squeeze_object = object()
+        dummy_objects.append(squeeze_object)
+        squeeze_object_name = str(id(squeeze_object))
+        slice_out_names = [squeeze_object_name]
+        if unsqueeze_idxs:
+            squeeze_out_names = [unsqueeze_object_name]
+        else:
+            squeeze_out_names = output_names
+        nodes.append(helper.make_node(
+            'Squeeze', [squeeze_object_name], squeeze_out_names,
+            axes=squeeze_idxs))
+
+    nodes.append(helper.make_node(
+        onnx_op_name, input_names, slice_out_names,
+        axes=axes, starts=starts, ends=ends))
+
+    return tuple(nodes)
+
+
+dummy_objects = []
```

Comment:
Wow, I didn't realize onnx-chainer has this weird hack. I hope we eventually re-implement the default naming scheme not to use `id`

posted in proxy, https://github.com/chainer/onnx-chainer/pull/72#discussion_r254586459

Reply:
Exactly, we should handle these IDs of converted node. I'll reorganize this node name creation.